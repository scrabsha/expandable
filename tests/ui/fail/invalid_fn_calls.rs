#[allow(unused_macros)]
#[expandable::expr]
macro_rules! call {
    // No argument, just a comma
    ($fn_name:ident()) => {
        $fn_name(,)
    };
}

#[allow(unused_macros)]
#[expandable::expr]
macro_rules! call_ {
    ($fn_name:ident()) => {
        $fn_name(,,)
    };
}

#[allow(unused_macros)]
#[expandable::expr]
macro_rules! call2 {
    ($fn_name:ident($arg1:expr, $arg2:expr)) => {
        $fn_name($arg1 $arg2)
    };
}

fn main() {}
