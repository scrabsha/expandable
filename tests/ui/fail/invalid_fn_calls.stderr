error: Potentially invalid expansion. Expected an identifier, a literal, `if`, a `[`, a `)`.
 --> tests/ui/fail/invalid_fn_calls.rs:6:18
  |
6 |         $fn_name(,)
  |                  ^

error: Potentially invalid expansion. Expected an identifier, a literal, `if`, a `[`, a `)`.
  --> tests/ui/fail/invalid_fn_calls.rs:14:18
   |
14 |         $fn_name(,,)
   |                  ^

error: Potentially invalid expansion. Expected `+`, `-`, `*`, `/`, `%`, `&`, `|`, `^`, `<<`, `>>`, `==`, `>`, `>=`, `<`, `<=`, `!=`, a `(`, `,`, a `)`, `.`.
  --> tests/ui/fail/invalid_fn_calls.rs:22:25
   |
22 |         $fn_name($arg1 $arg2)
   |                         ^^^^
